### PR TITLE
Automate API Docs Publication

### PR DESCRIPTION
**Pull request by tobiasschweizer**
Automates generation and publication of API docs to github-pages.

### CODE REVIEW

**Comment by tobiasschweizer** [comment]
@kilchenmann I could set up a github-ci job using https://github.com/tsunematsu21/actions-publish-gh-pages to generate the docs and publish them to gh-pages.

You can see the result here: https://dasch-swiss.github.io/dsp-js-lib/

https://github.com/dasch-swiss/dsp-js-lib/blob/dcf7ff721f970c7b564d931347ea21fdc5ed1797/.github/workflows/main.yml#L117-L136

At the moment, the conditions are commented out so we can test this.
I think this should only happen on release, RCs excluded. 

Not sure how we could have versioned API docs, I will do a bit of research.

**Comment by kilchenmann** [comment]
> @kilchenmann I could set up a github-ci job using https://github.com/tsunematsu21/actions-publish-gh-pages to generate the docs and publish them to gh-pages.
> 
> You can see the result here: https://dasch-swiss.github.io/dsp-js-lib/
> 
> https://github.com/dasch-swiss/dsp-js-lib/blob/dcf7ff721f970c7b564d931347ea21fdc5ed1797/.github/workflows/main.yml#L117-L136
> 
> At the moment, the conditions are commented out so we can test this.
> I think this should only happen on release, RCs excluded.
> 
> Not sure how we could have versioned API docs, I will do a bit of research.

Yes, this is exactly the same we do in DSP-API and DSP-Docs to publish Documentation to github pages.

**Comment by tobiasschweizer** [comment]
I made a [feature request](https://github.com/tsunematsu21/actions-publish-gh-pages/issues/19).

**Comment by kilchenmann** [comment]
> I made a [feature request](https://github.com/tsunematsu21/actions-publish-gh-pages/issues/19).

Ok. But maybe there are better Github actions on the [marketplace](https://github.com/marketplace?type=actions) to [deploy documentation to github-pages](https://github.com/marketplace?type=actions&query=github+pages) with different versions.

**Comment by tobiasschweizer** [comment]
@kilchenmann We solved the problem by adding versioned directory where the docs are built in: [https://dasch-swiss.github.io/dsp-js-lib/1_0_0/index.html](https://dasch-swiss.github.io/dsp-js-lib/1_0_0/index.html)

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -113,3 +113,26 @@ jobs:
         uses: lakto/google-chat-action@main
         with:
           url: ${{ secrets.GOOGLE_CHAT_DSP_RELEASES_WEBHOOK_URL }}
+
+  publish-docs:
+    name: Build and publish API docs
+    runs-on: ubuntu-latest
+    #needs: [build]
+    #if: github.event_name == 'release' && startsWith(github.ref, 'refs/tags')
```
Comment (tobiasschweizer) [inline .github/workflows/main.yml:121]: @kilchenmann any idea how to exclude RCs here?

Comment (kilchenmann) [reply .github/workflows/main.yml:121]: you could add it after publishing to npm. There's a section for RC and one for real releases. Or you could copy the if statement from there: `- if: "!github.event.release.prerelease"`

Comment (tobiasschweizer) [reply .github/workflows/main.yml:121]: So then it would build and publish the docs only for releases that are **not** flagged as prereleases, right?

Comment (kilchenmann) [reply .github/workflows/main.yml:121]: Yes

**Comment by tobiasschweizer** [comment]
@kilchenmann I made an `index.html` that lists the versions dynamically, see https://dasch-swiss.github.io/dsp-js-lib (https://github.com/dasch-swiss/dsp-js-lib/blob/gh-pages/index.html).

**Review by mdelez — APPROVED** [review]
Neat!

**Review by kilchenmann — APPROVED** [review]

**Pull request merged**